toolchain/sem_ir/value_stores.h

```diff
@@ -43,6 +43,48 @@ class NodeStore {
   ValueStore<NodeId, Node> values_;
 };
 
+// Provides a ValueStore-like interface for names. This is currently a wrapper
+// around a string store that has no state of its own.
+class NameStore {
+ public:
+  NameStore(ValueStore<StringId>* strings) : strings_(strings) {}
+
+  // Adds a string name, returning an ID to reference it.
+  auto Add(StringId string_id) -> NameId { return NameId(string_id.index); }
```

Comment:
This feels a bit odd: it's not really adding things, it's just wrapping the type. Had you considered a constructor on NameId?

Reply:
The intended abstraction here is that `NameStore` is a `ValueStore` for names, that happens to not need to actually store any state at all. I don't mind adding a `NameId` constructor, but it does break that abstraction a little bit by encoding into the API the fact that we can convert between `NameId`s and `StringId`s without a `NameStore`.

Would a different name for this function resolve your feeling of oddness, or do you think we should be more directly exposing the conversion between `NameId` and `StringId`?

Reply:
A different name ("Wrap"?) would help, although maybe also renaming `NameStore` -> `Names` or similar may help too. As mentioned, it's actually the "ValueStore-like" aspect that I'm concerned about this being treated as a `ValueStore`.

If you're okay with #3352, you could also consider merging this with the `StringStore` there, just having `NameId` one more thing that it can toss around. Then, versus not doing "any additional explicit storage", it's merged with the rest in practice.

The converse of the above -- as I'm thinking more -- is if you don't want to merge with `StringStore` from #3352, and given the planned separation of `SharedValueStore` to be per-compilation-unit, then this probably belongs in `SemIR`. But I'd still then prefer this to be split a little more from `Store` objects which actually store their data, so that we don't end up with a new string store by accident.

Reply:
Renamed to `NameStoreWrapper` to align with `StringStoreWrapper`.